src/Options/EncapsulatedOptions.php

```diff
@@ -61,6 +91,10 @@
      */
     final protected function set($option, $value)
     {
+        if (is_object($value) || is_resource($value)) {
```

Comment:
TODO: This value type checking needs to be extended to recursively check array values and defaults.